### PR TITLE
[Beam-520] Update Python SDK example tests to use assert_that

### PR DESCRIPTION
**Pull request by venbijjam**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:
- [ ] Make sure the PR title is formatted like:
  `[BEAM-<Jira issue #>] Description of pull request`
- [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
     Travis-CI on your fork and ensure the whole test matrix passes).
- [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
     number, if there is one.
- [ ] If this contribution is large, please file an Apache
     [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
Thanks! 

You might want to rebase on https://github.com/apache/incubator-beam/pull/1110 (or the lastest python-sdk tip) to get the tests passing.

**Comment by venbijjam** [comment]
Hi
I rebased and Travis Build# 4754 is now in the pipeline.

But, we need to cancel the older Travis Build# 4744 because it errored out in 1 build task due to #1110..

**Comment by venbijjam** [comment]
@robertwb, could you merge this into Beam?

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py [hunk 1]
```diff
@@ -18,54 +18,43 @@
 """Test for the multiple_output_pardo example."""
 
 import logging
-import re
-import tempfile
 import unittest
 
+import apache_beam as beam
 from apache_beam.examples.cookbook import multiple_output_pardo
+#from apache_beam.transforms.util import assert_that
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: Please remove the commented out import statements.

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py [hunk 2]
```diff
@@ -18,54 +18,43 @@
...
-      for line in result_file:
-        match = re.search(r'([A-Za-z]+): ([0-9]+)', line)
-        if match is not None:
-          results.append((match.group(1), int(match.group(2))))
-    return results
+  SAMPLE_TEXT = 'A whole new world\nA new point'
+  text_len = len(' '.join(SAMPLE_TEXT.split('\n')))
+  SAMPLE_TEXT_Iterable = [SAMPLE_TEXT]
+  EXPECTED_SHORT_WORDS = ['A: 2', 'new: 2']
+  EXP_WORDS = ['point: 1', 'whole: 1', 'world: 1']
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: Variable naming is not consistent within this file and with other tests. Could you fix that, for example:

`EXP_WORDS` -> `EXPECTED_WORDS`
`SAMPLE_TEXT_Iterable` -> `SAMPLE_TEXT_ITERABLE`
`res _sh_wrd` -> `res_short_words`
`res_cnt` -> `res_count`

Maybe even spell out `res` to `results`.

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py [hunk 3]
```diff
@@ -18,54 +18,43 @@
...
-
-    short_words = self.get_wordcount_results(
-        result_prefix + '-short-words-00000-of-00001')
-    self.assertEqual(sorted(short_words), sorted(self.EXPECTED_SHORT_WORDS))
-
-    words = self.get_wordcount_results(result_prefix + '-words-00000-of-00001')
-    self.assertEqual(sorted(words), sorted(self.EXPECTED_WORDS))
-
+    p = beam.Pipeline('DirectPipelineRunner')
+    sample_text = p | beam.Create(self.SAMPLE_TEXT_Iterable)
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: `beam.Create([self.SAMPLE_TEXT])` could be a simpler way of expressing this, without using another variable.

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py [hunk 4]
```diff
@@ -18,54 +18,43 @@
...
+          )
+    res_cnt = (res.tag_character_count
+               | 'pair_with_key' >> beam.Map(lambda x: ('chars_temp_key', x))
+               | beam.GroupByKey()
+               | 'count chars' >> beam.Map(lambda (_, counts): sum(counts)))
+    res_words = (res.words
+                 | 'count words' >> multiple_output_pardo.CountWords())
+    res_sh_wrd = (res.tag_short_words
+                  | 'count short words' >> multiple_output_pardo.CountWords())
+    beam.assert_that(res_words, beam.equal_to(self.EXP_WORDS))
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: You added labels to the other asserts, please add one to this one as well.

File: sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py [hunk 5]
```diff
@@ -18,10 +18,10 @@
 """Test for the mergecontacts example."""
 
 import logging
-import tempfile
+import re
 import unittest
 
-from apache_beam.examples.cookbook import mergecontacts
```
Comment (robertwb) [inline sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py]: It's not much of a test of mergecontacts if mergecontacts isn't even imported. 

I suggest we leave this test as is, as reading and writing are a core part of the functionality. (It doesn't do much else.)

Comment (robertwb) [inline sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py]: (Similarly for most of the rest of the cookbook examples, e.g. group_with_coder_test. If there are non-trivial transforms, we could add tests for those in isolation.)

Changing the tests of functionality that has nothing to do with IO, however, is very useful.

File: sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py [hunk 6]
```diff
@@ -101,22 +99,61 @@ def normalize_tsv_results(self, tsv_data):
...
+    #(self.normalize_tsv_results('\n'.join(result_tsv_lines)),
+    beam.assert_that(result_tsv_lines,
+                     beam.equal_to([self.EXPECTED_TSV]),
+                     label='assert:tag_tsv_results')
+    beam.assert_that(num_luddites, beam.equal_to([self.EXPECTED_LUDDITES]),
+                     label='assert:tag_num_luddites')
+    beam.assert_that(num_writers, beam.equal_to([self.EXPECTED_WRITERS]),
+                     label='assert:tag_num_writers')
+    beam.assert_that(num_nomads, beam.equal_to([self.EXPECTED_NOMADS]),
+                     label='assert:tag_num_nomads')
```
Comment (robertwb) [inline sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py]: Always call p.run() to run the asserts. 

Alternatively, do
```
with Pipeline(...) as p:
    ...
```

which calls run on exiting the block.

**Comment by aaltay** [comment]
@venbijjam the last jenkins error is unrelated to your changes.

As suggested in one of the comments, would you like to keep only `multiple_output_pardo_test.py` in this PR and move others to another PR. That way you can make progress and we con continue reviewing more files you are planning to change.

**Comment by aaltay** [comment]
Hi @venbijjam any progress on this one? Please let me know if you need help making progress. I will be happy to help.

**Comment by venbijjam** [comment]
@aaltay i worked on 3 files...but the jenkins build failed when i committed the third file..travis was successful...

**Comment by aaltay** [comment]
@venbijjam 

Last Jenkins failure was unrelated to your change. (It was a failure for another reason, that is fixed now.) We can ignore that failure.

Do you need help with addressing the reviewer comments? My suggestion is to not convert mergecontacts test and leave the two other tests that you converted. We can rebase again to bring your changes to a fresh status and quickly review the two files you converted.

Please let me know if I could help with the above plan. Thank you again for contributing.

**Comment by aaltay** [comment]
Hi @venbijjam, I want to check how this PR is going. If you need help please feel free to reach me. Thank you!

**Pull request closed without merging**